src/testdir/test_ins_complete.vim

```diff
@@ -2764,24 +2767,73 @@ func Test_complete_fuzzy_match()
   call assert_equal('你好 他好', getline('.'))
 
   " issue #15526
-  set completeopt=fuzzy,menuone,menu,noselect
+  set completeopt=menuone,menu,noselect
   call setline(1, ['Text', 'ToText', ''])
   call cursor(3, 1)
   call feedkeys("STe\<C-X>\<C-N>x\<CR>\<Esc>0", 'tx!')
   call assert_equal('Tex', getline('.'))
 
-  " clean up
-  set omnifunc=
   bw!
   bw!
-  set complete& completeopt&
-  autocmd! AAAAA_Group
-  augroup! AAAAA_Group
-  delfunc OnPumChange
-  delfunc Omni_test
-  delfunc Comp
-  unlet g:item
-  unlet g:word
+  set completeopt& cfc& cpt&
+endfunc
+
+func Test_cfc_with_longest()
+  new
+  set completefuzzycollect=k,f,l
+  set completeopt=menu,menuone,longest,fuzzy
+
+  " keyword
+  exe "normal ggdGShello helio think h\<C-X>\<C-N>\<ESC>"
+  call assert_equal("hello helio think hel", getline('.'))
+
+  " kdcit
+  call writefile(['help'], 'test_keyword.txt', 'D')
```

Comment:
The same issue too, if we just add an extra space, the test will not pass.

```
  call writefile([' help'], 'test_keyword.txt', 'D')
```

Reply:
fixed

Reply:
In the real scenario, my dictionary file is placed in a folder of vim, not the current folder. For example:

```vim
augroup c_lang_context
    autocmd!
    autocmd FileType c setlocal complete+=k~/.vim/test/c_keywords.c
augroup END
```

If so, when I press `<C-n>`, vim will freeze.

Reply:
could you give me some content of keywords.c just like normal ? typedef etc?

Reply:
reproduced wait for a while

Reply:
Great, now everything works as expected. If everything goes well, like adding more test cases to improve code coverage and the CI passes, do you think it's time to open this PR as ready? What do you think, or you want to add more features to this PR in addition to resolve other corner case bugs?

Reply:
no more are being added at this time. it's working roughly as required, what needs to be done is to add more tests to catch more edge cases.

Reply:
Okay, got it, thanks again for your work, looking forward to using this great feature in the release version soon. 😄

Reply:
See your latest commit, Sorry to bother you, is the PR ready to open now 👀 ? Looking forward to this feature 😄. Also, Happy New Year in advance 🎉.

Reply:
I'll clean it up later.  thanks you too.